api/app/hfi/hfi.py

```diff
@@ -92,12 +89,14 @@ def calculate_hfi_results(fire_centre_id: int,  # pylint: disable=too-many-local
 def calculate_max_intensity_group(mean_intensity_groups: List[Optional[float]]):
     """ Returns the highest intensity group from a list of values """
     valid_mean_intensity_groups = list(filter(None, mean_intensity_groups))
-    return None if len(valid_mean_intensity_groups) == 0 else max(mean_intensity_groups)
+    # TODO: establish, if there are any invalid intensity groups, do we return None?
+    return None if len(valid_mean_intensity_groups) == 0 else max(valid_mean_intensity_groups)
```

Comment:
@conbrad should this return none when `len(valid_mean_intensity_groups) != num_prep_days` ?